### PR TITLE
[jsx/(Filter.js FilterableDataTable.js DataTable.js)] Browser console error cleanup

### PR DESCRIPTION
**Pull request by maltheism**
## Brief summary of changes

Updated the propTypes to correspond with the correct "type" of the data being passed to the components. Updated DataTable.js to have a key be passed using React.cloneElement.

#### Testing instructions (if applicable)

1. Visit the candidate_list module and view the browser console.

edit: Make sure the sandbox value is set to 1.
```
<config>
    <!-- set to 1 if development environment -->
    <dev>
        <sandbox>1</sandbox>
    </dev>
...
</config>
```

#### Link(s) to related issue(s)

* Resolves #  (Reference the issue this fixes, if any.)

#6379

### CODE REVIEW

File: jsx/DataTable.js [hunk 1]
```diff
@@ -440,7 +440,7 @@ class DataTable extends Component {
                 // Note: Can't currently pass a key, need to update columnFormatter
                 // to not return a <td> node. Using createFragment instead.
                 // let key = 'td_col_' + j;
-                curRow.push(cell);
+                curRow.push(React.cloneElement(cell, {key: 'td_col_' + j}));
```
Comment (maltheism) [inline jsx/DataTable.js:443]: This line solves `[Error] Warning: Each child in a list should have a unique "key" prop.` and the React.cloneElement just clones the cell with passing a key to it. The prior existing key was null.

**Comment by maltheism** [comment]
@kongtiaowang @johnsaigle @driusan do either of you happen to know why travis is throwing `Error: Class 'FirefoxProfile' not found` errors?

File: jsx/DataTable.js [hunk 2]
```diff
@@ -440,7 +440,7 @@ class DataTable extends Component {
                 // Note: Can't currently pass a key, need to update columnFormatter
                 // to not return a <td> node. Using createFragment instead.
                 // let key = 'td_col_' + j;
```
Comment (HenriRabalais) [inline jsx/DataTable.js:442]: This line doesn't seem necessary any more with the changes made!

**Comment by maltheism** [comment]
Thanks @HenriRabalais, I removed the unnecessary comments.

**Comment by kongtiaowang** [comment]
https://github.com/aces/Loris/pull/6392

**Comment by jesscall** [comment]
Nothing appears in console, I imagine this is the expected behaviour. Not sure how else this PR should be tested?

Screen Shot 2020-04-16 at 2 08 40 PM


I noticed something weird though when on your branch. There is an awkward white space / gap between the top LORIS bar and the breadcrumbs... (**not** unique to the candidate_profile page)

Screen Shot 2020-04-16 at 2 08 07 PM

**Comment by maltheism** [comment]
@jesscall you can test viewing the console while not on the PR to see the error messages. Hmm, I'm not sure why the white gap exists from the code changes I did. I'll take a look.

**Comment by maltheism** [comment]
@jesscall I'm not seeing the white space / gap like in your picture when viewing my PR on chrome. Could you try testing again and clearing your browser cache?

**Comment by jesscall** [comment]
@maltheism I do not get console errors when on 23.0-release branch. Maybe someone else can test this and confirm if it indeeds resolves the original issue?

**Comment by maltheism** [comment]
@jesscall are you testing on react.development.js or react.production.js?

**Comment by maltheism** [comment]
@ridz1208 can you test?

**Comment by kongtiaowang** [comment]
I think it is ready.

**Pull request merged**